R/helper_lm_robust_fit.R

```diff
@@ -272,54 +226,54 @@ lm_robust_fit <- function(y,
   if (return_fit) {
     if (iv_second_stage && !fes) {
       if (weighted) {
-        return_list[["fitted.values"]] <- as.matrix(X_first_stage_unweighted %*% fit$beta_hat)
+        return_list[["fitted.values"]] <- as.matrix(data[["X_first_stage_unweighted"]] %*% fit$beta_hat)
       } else {
         return_list[["fitted.values"]] <- as.matrix(fitted.values)
       }
-    } else if ((se_type == "CR2" && weighted)) {
+    } else if (se_type == "CR2" && weighted && (iv_first_stage || !return_unweighted_fit)) {
```

Comment:
This logic is confusing, roughly six different flags are interacting. Add a todo to refactor after this next version ships.

Reply:
Yes, this is absolutely brutal.